planner/core/logical_plan_builder.go

```diff
@@ -362,13 +367,32 @@ func (b *PlanBuilder) buildResultSetNode(ctx context.Context, node ast.ResultSet
 			return nil, err
 		}
 
+		var dbName4DerivedTable model.CIStr
+		if isSubQuery {
+			// change the derived table's database name.
+			asName := x.AsName
+			x.AsName = model.NewCIStr("")
+			var sb strings.Builder
+			// (select * from t) as t1 => db name as (select * from t)
+			err = x.Restore(format.NewRestoreCtx(0, &sb))
+			if err != nil {
+				return nil, err
+			}
+			dbName4DerivedTable = model.NewCIStr(sb.String())
+			x.AsName = asName
+		}
```

Comment:
Why do we need this code block? If strange that we add such a database name.

Reply:
mainly target at this cast：`select * from db1.t as t1, (select * from t) as t1;`

the first half‘s schema is db1.t1.a
the second half's schema is currentDB(db1).t1.a

so this two will lead to ambiguous problems.

`select * from db1.t as t1, db2.t as t1;`

while this case can succeed, because

the first half‘s schema is db1.t1.a
the second half's schema is db2.t1.a